docs/keypairs.md

```diff
@@ -0,0 +1,86 @@
+    [--user-domain <user-domain>]
+    [--project <project>]
+    [--project-domain <project-domain>]
+    [--limit <limit>]
+    [--marker <marker>]
+```
+
+#### Create a Keypair
+
+Before launching an instance, you must add a public key to the Compute service.
+
+``` shell
+$ openstack keypair create KEY_NAME > MY_KEY.pem
+```
+This command generates a key pair with the name that you specify for KEY_NAME, writes the private key to the .pem file that you specify, and registers the public key to the Nova database.
+
+#### Import a Keypair
+
+If you have already generated a key pair and the public key is located at ~/.ssh/id_rsa.pub, run the following command to upload the public key.
+
+``` shell
+$ openstack keypair create --public-key ~/.ssh/id_rsa.pub KEY_NAME
+```
+
+This command registers the public key at the Nova database and names the key pair the name that you specify for KEY_NAME
+
+#### Delete a Keypair
+
+``` shell
+$ openstack keypair delete
+    [--user <user>]
+    [--user-domain <user-domain>]
+    <key>
+    [<key> ...]
+```
+
+#### Show Keypair Details
+
+``` shell
+$ openstack keypair show
+    [--public-key]
+    [--user <user>]
+    [--user-domain <user-domain>]
+    <key>
+```
+
+# Nova Keypairs
+
+Read more about Nova keypairs using the [upstream docs](https://specs.openstack.org/openstack/nova-specs/specs/kilo/implemented/keypair-x509-certificates.html).
+
+Nova keypairs are mostly used by Linux guests to handle user authentication via SSH public key authentication without incurring in the management and security overhead that passwords require.
+
+Public keys are provided to the guests as part of the metadata and included in the guest configuration by tools like cloud-init.
+
+Windows operating systems don’t support natively SSH and thus authentication requires the usage of passwords unless the image deployer chooses to include a 3rd party unsupported SSH service port, which implies incurring in potential security and support issues.
+
+Windows supports natively password-less authentication for WinRM by using X509 certificates, including PowerShell remoting.
+
+X509 certificates are used by WinRM in a way which can be considered consistent with the usage of SSH keys on Linux, as both are based on public / private keypairs. This is limited by the lack of native availability of SSH on Windows.
+
+Certificates can be generated by the user or by Nova (like for the SSH keypair case). Self-signed certificates are accepted.
+
+####Use Cases
+
+End User will be able to connect to a Windows instance without requiring a password.
+
+Like for the SSH keypair case, the use has the choice to generate the X509 certificate and pass the public X509 to Nova (recommended) or let Nova generate and return the certificate including the private key.
+
+####Generating an x509 Keypair
+
+Generating an x509 keypair via CLI and saving the private certificate locally (in PKCS#12 format):
+
+``` shell
+$ nova keypair-add keypair1 –keypair-type x509
```

Comment:
can we generate this with the `openstack` command?

Reply:
I am looking into that, we can specify type of certificate in create. I am confirming that x509 is included.